pkg/registry/bundle.go

```diff
@@ -210,19 +211,21 @@ func (b *Bundle) cache() error {
 	}
 
 	if b.crds == nil {
-		b.crds = []*apiextensions.CustomResourceDefinition{}
+		b.crds = []*v1beta1.CustomResourceDefinition{}
 	}
 	for _, o := range b.Objects {
 		if o.GetObjectKind().GroupVersionKind().Kind == "CustomResourceDefinition" {
-			crd := &apiextensions.CustomResourceDefinition{}
-			// Marshal Unstructured and Unmarshal as CustomResourceDefinition. FromUnstructured has issues
+			// Marshal Unstructured and Decode as CustomResourceDefinition. FromUnstructured has issues
 			// converting JSON numbers to float64 for CRD minimum/maximum validation.
-			bytes, err := o.MarshalJSON()
+			cb, err := o.MarshalJSON()
 			if err != nil {
 				return err
 			}
-			if err := json.Unmarshal(bytes, &crd); err != nil {
-				return err
+			crd := &v1beta1.CustomResourceDefinition{}
+			dec := serializer.NewCodecFactory(Scheme).UniversalDeserializer()
```

Comment:
This will throw an error if a CRD version created in the future wasn't registered with `Scheme`. This is could be considered a feature ("we don't explicitly support that CRD version yet, throw an error"); simply using `json.Unmarshal` may cause silent marshaling errors like the one I describe in the PR comments. Thoughts?